pkg/http/otel.go

```diff
@@ -40,6 +41,22 @@ func LogTraceID(next http.Handler) http.Handler {
 	})
 }
 
+// BaggageToSpanAttributes extracts all baggage members from the context
+// and adds them as attributes to the current span.
+func BaggageToSpanAttributes(next http.Handler) http.Handler {
```

Comment:
there is a standard way to do it via span processor and there is an existing implementation - check out https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/processors/baggagecopy:


```go
   import (
       "go.opentelemetry.io/contrib/processors/baggagecopy"
   )

	tracerProvider := sdktrace.NewTracerProvider(
		sdktrace.WithSpanProcessor(
			baggagecopy.NewSpanProcessor(
				func(m baggage.Member) bool {
					return true
				},
			),
		),
       sdktrace.WithBatcher(traceExporter...
```